### PR TITLE
[#19][#25][API] As a user, after upload a keywords file, I can see the list of keywords on the screen. [API] As a user, I can search across all the reports

### LINKED ISSUE #19: [API] As a user, after upload a keywords file, I can see the list of keywords on the screen
## Why

After the user uploads a keywords file, we need to show a list of keywords from that file on the screen.

## Acceptance Criteria

to be provided

### LINKED ISSUE #25: [API] As a user, I can search across all the reports
## Why

After getting all reports from the keywords list, we need to support a search feature, so the user can search and find the report they want.

## Acceptance Criteria

to be provided

### PR DESCRIPTION
**Pull request by suho**
Resolves #19
Resolves #25

## What happened 👀

- After the user uploads a keywords file, we need to show a list of keywords from that file on the screen.
- After getting all reports from the keywords list, we need to support a search feature, so the user can search and find the report they want.
 
## Insight 📝

- Create API to support getting a list of keywords of a user
- Support paging with 2 parameters: page[number], page[size]
-  Also support filter parameter: keyword

Endpoint: `{{base_url}}/api/v1/keywords?keyword=So&page[number]=1&page[size]=1`
 
## Proof Of Work 📹

Screen Shot 2022-01-14 at 23 21 16

### CODE REVIEW

File: spec/controllers/api/v1/keywords_controller_spec.rb [hunk 1]
```diff
@@ -3,10 +3,79 @@
 require 'rails_helper'
 
 RSpec.describe API::V1::KeywordsController, type: :controller do
-  describe 'POST#create', api_authentication: :user do
+  describe 'GET#index' do
+    context 'given a request without params' do
```
Comment (hanam1ni) [inline spec/controllers/api/v1/keywords_controller_spec.rb]: Should we add a test to expect that it return a valid keywords?

File: spec/controllers/api/v1/keywords_controller_spec.rb [hunk 2]
```diff
@@ -3,10 +3,79 @@
...
+          api_sign_in
+          get :index, params: { page: { number: 1000, size: 2 } }
+
+          expect(response).to have_http_status(:not_found)
+        end
+      end
+    end
+
+    context 'given a request with search keyword params' do
+      it 'returns expected status' do
```
Comment (hanam1ni) [inline spec/controllers/api/v1/keywords_controller_spec.rb]: For this case, we should test if it return a valid keyword

Comment (suho) [reply spec/controllers/api/v1/keywords_controller_spec.rb]: Good catch! I forgot to add test cases to check about the response content, will add more test cases about this

File: spec/controllers/api/v1/keywords_controller_spec.rb (hunk 1 above)
Comment (suho) [reply spec/controllers/api/v1/keywords_controller_spec.rb]: Fixed in [`ddb5fab` (#55)](https://github.com/suho/google-search-ruby/pull/55/commits/ddb5fabf379d63d5c7e6ad4edc49abbde8fe2297)

File: spec/controllers/api/v1/keywords_controller_spec.rb (hunk 2 above)
Comment (suho) [reply spec/controllers/api/v1/keywords_controller_spec.rb]: Fixed in [`ddb5fab` (#55)](https://github.com/suho/google-search-ruby/pull/55/commits/ddb5fabf379d63d5c7e6ad4edc49abbde8fe2297)

**Review by hanam1ni — APPROVED** [review]

File: app/serializers/keyword_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class KeywordSerializer
+  include JSONAPI::Serializer
+  attributes :keyword,
```
Comment (junan) [inline app/serializers/keyword_serializer.rb:6]:
```suggestion

  attributes :keyword,
```

File: app/serializers/link_serializer.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class LinkSerializer
+  include JSONAPI::Serializer
+  attributes :url,
```
Comment (junan) [inline app/serializers/link_serializer.rb:6]:
```suggestion

  attributes :url,
```

**Review by junan — APPROVED** [review]

**Pull request merged**